### PR TITLE
Fixes #119

### LINKED ISSUE #119: amplicon pre-preparation file generator notebook: generate warning if invalid characters found in sample_name
Currently when a sample_name containing a character that is not acceptable in Qiita goes through the amplicon notebook, there is no warning produced and the user can go on to making a pre-preparation file that includes invalid characters. It would be beneficial to add a warning that does not allow the user to move forward unless all sample_name characters are confirmed valid.

For example, if someone names a BLANK sample BLANK.L'OREAL.2.4E, no warning or error will appear (but ' is not an acceptable character in a sample_name).

Thanks!

### PR DESCRIPTION
**Pull request by charles-cowart**
Added sanitization functions for sample-names in plate-map files.

### CODE REVIEW

File: metapool/metapool.py [hunk 1]
```diff
@@ -171,6 +171,52 @@ def sum_lanes(multi_lane_df, lanes_to_sum):
...
+    '''
+
+    # remove any leading and/or trailing whitespace from sample-names only.
+    plate_df['tmp'] = plate_df.Sample.str.strip()
+
+    # note which columns needed cleaning. We'll be returning this info to the
+    # user.
+    plate_df['changed'] = plate_df['Sample'] != plate_df['tmp']
+    repaired_names = list(
+        plate_df.loc[plate_df['Sample'] != plate_df['tmp']]['Sample'])
```
Comment (antgonza) [inline metapool/metapool.py:204]:
```suggestion
    repaired_names = plate_df.loc[plate_df['changed'].to_list()
```

**Review by antgonza — CHANGES_REQUESTED** [review]
Some questions.

File: metapool/metapool.py [hunk 2]
```diff
@@ -171,6 +171,52 @@ def sum_lanes(multi_lane_df, lanes_to_sum):
...
+    Taken from Qiita's util.py:get_invalid_sample_names().
+
+    :param plate_df: A DataFrame representing a plate-map file.
+    :return: List of invalid sample-names.
+    '''
+
+    # convert all values to strings to handle instances when values like
+    # NaN are used as sample-names. Currently 'NaN' is an acceptable name.
+    sample_names = [str(x) for x in list(plate_df['Sample'])]
+    valid_chars = set(ascii_letters + digits + '.' + '-')
```
[the remark this answers is not in the document]
Comment (charles-cowart) [reply metapool/metapool.py:186]: Not sure I see the benefit of having valid_chars be valuable. It makes it easier to change what characters are valid if they are hidden behind the calling function. Incidentally, all ASCII characters + digits + '.' and '-' sounds reasonable? I based it on one of Jose's old functions + adding '-' so a lot of legacy tests didn't break. (I figure if '-' were in the legacy tests they could reasonably be expected to appear in actual output.)

[the remark this answers is not in the document]
Comment (antgonza) [reply metapool/metapool.py:186]: All that sounds good.

**Comment by charles-cowart** [comment]
@antgonza bugfix is in. all tests are passing again. Ready for re-review!

**Review by antgonza — APPROVED** [review]

**Pull request merged**